projects/js-packages/shared-extension-utils/src/block-editor-actions.js

```diff
@@ -0,0 +1,55 @@
+import { dispatch } from '@wordpress/data';
+import domReady from '@wordpress/dom-ready';
+
+const JETPACK_EDITOR_ACTION = 'jetpack-editor-action';
+
+/**
+ * Get the Jetpack Editor action from the URL.
+ *
+ * @return {string | null} The Jetpack Editor action.
+ */
+export function getJetpackEditorAction() {
+	const url = new URL( window.location.href );
+
+	return url.searchParams.get( JETPACK_EDITOR_ACTION );
+}
+
+/**
+ * Remove the Jetpack Editor action from the URL.
+ *
+ */
+export function removeJetpackEditorAction() {
+	const url = new URL( window.location.href );
+	url.searchParams.delete( JETPACK_EDITOR_ACTION );
+	window.history.replaceState( null, '', url.toString() );
+}
+
+/**
+ * Handle the Jetpack Editor action.
+ *
+ * @param {{sidebarToOpen?: string; onAction?: (action:string) => void; removeQueryArg?: boolean}} args - Arguments.
+ */
+export function handleJetpackEditorAction( {
+	sidebarToOpen = 'jetpack-sidebar/jetpack',
+	removeQueryArg = true,
+	onAction,
+} ) {
+	domReady( () => {
+		const action = getJetpackEditorAction();
+		if ( action ) {
```

Comment:
I think we should pass in the action param we're looking for and a callback. If `action` equals the value we call the callback.

Reply:
Done in f19e968463a2027f10d7a48c35b84c2495892b46